app/src/organisms/ChooseProtocolSlideout/index.tsx

```diff
@@ -193,164 +192,273 @@ export function ChooseProtocolSlideoutComponent(
+                      as="h3"
+                      fontWeight={TYPOGRAPHY.fontWeightSemiBold}
+                    >
+                      {t('protocol_details:csv_file')}
+                    </StyledText>
+                    <StyledText as="p">
+                      {t('protocol_details:csv_required')}
+                    </StyledText>
+                  </Flex>
+                  {runtimeParam.file == null ? (
+                    <UploadInput
+                      uploadButtonText={t('protocol_details:choose_file')}
+                      onUpload={(file: File) => {
+                        const clone = runTimeParametersOverrides.map(
+                          parameter => {
+                            if (
+                              runtimeParam.variableName ===
+                              parameter.variableName
+                            ) {
+                              return {
+                                ...parameter,
+                                file: { file: file },
+                              }
+                            }
+                            return parameter
+                          }
+                        )
+                        setRunTimeParametersOverrides?.(clone)
+                      }}
+                      dragAndDropText={
+                        <StyledText as="p">
+                          <Trans
+                            t={t}
+                            i18nKey="shared:drag_and_drop"
+                            components={{
+                              a: (
+                                <LinkComponent
+                                  color={COLORS.blue55}
+                                  role="button"
+                                  to={''}
+                                />
+                              ),
+                            }}
+                          />
+                        </StyledText>
+                      }
+                    />
+                  ) : (
+                    <FileCard
+                      error={error}
+                      fileRunTimeParameter={runtimeParam}
+                      runTimeParametersOverrides={runTimeParametersOverrides}
+                      setRunTimeParametersOverrides={
+                        setRunTimeParametersOverrides
+                      }
+                    />
+                  )}
+                </Flex>
+              )
+            }
+          }
         )
-      }
-    }) ?? null
+      : null
 
   const resetRunTimeParameters = (): void => {
-    setRunTimeParametersOverrides(
-      runTimeParametersOverrides?.map(parameter => ({
-        ...parameter,
-        value: parameter.default,
-      }))
+    const clone = runTimeParametersOverrides.map(parameter =>
+      parameter.type === 'csv_file'
+        ? { ...parameter, file: null }
+        : { ...parameter, value: parameter.default }
     )
+    setRunTimeParametersOverrides(clone as RunTimeParameter[])
```

Comment:
nit, maybe you can define clone's type instead

```suggestion
    const clone :  RunTimeParameter[] = runTimeParametersOverrides.map(parameter =>
      parameter.type === 'csv_file'
        ? { ...parameter, file: null }
        : { ...parameter, value: parameter.default }
    )
    setRunTimeParametersOverrides(clone)
```

Reply:
Unfortunately, ts is failing unless I explicitly cast `clone` in the state setter. I am going to create a followup PR that provides a larger-scale type update for RTPs that should avoid these conflicts moving forward (per my standup update).

Reply:
Cool ya i think the amount of times we have to type stuff throughout the RTPs code is a bit messy so that'd be awesome to clean it up in a followup at some point.

Reply:
going to mark this convo as resolved and address the types in my next PR.